components/PullRequests/PrStateAuthorIcon.tsx

```diff
@@ -0,0 +1,63 @@
+import { GitMergeIcon, GitPullRequestClosedIcon, GitPullRequestIcon } from "@primer/octicons-react";
+import { Avatar } from "components/atoms/Avatar/avatar-hover-card";
+
+function getPullRequestStateIcon(
+  state: DbRepoPREvents["pr_state"],
+  isDraft: DbRepoPREvents["pr_is_draft"],
+  isMerged: DbRepoPREvents["pr_is_merged"]
+) {
+  const size = 14;
+
+  switch (true) {
+    case state === "open" && !isDraft:
+      return <GitPullRequestIcon aria-label="open pull request" size={size} className="text-white" />;
+    case state === "open" && isDraft:
+      return <GitPullRequestIcon size={size} aria-label="draft pull request" className="text-white" />;
+    case state === "closed" && !isMerged:
+      return <GitPullRequestClosedIcon size={size} aria-label="closed pull request" className="text-white" />;
+    case state === "closed" && isMerged:
+      return <GitMergeIcon size={size} aria-label="merged pull request" className="text-white" />;
+  }
+}
+
+export const PrStateAuthorIcon = ({
+  state,
+  isDraft,
+  isMerged,
+  author,
+}: {
+  state: string;
```

Comment:
Restrict to valid states

```suggestion
  state: "open" | "closed";
```

Reply:
Good catch.